File: decompiled/sources/com/hjq/widget/layout/RatioFrameLayout.java
package com.hjq.widget.layout;

import android.content.Context;
import android.content.res.TypedArray;
import android.text.TextUtils;
import android.util.AttributeSet;
import android.view.View;
import android.view.ViewGroup;
import android.widget.FrameLayout;
import androidx.constraintlayout.solver.widgets.analyzer.BasicMeasure;
import com.hjq.widget.R;

public final class RatioFrameLayout extends FrameLayout {
    private float mHeightRatio;
    private float mWidthRatio;

    public RatioFrameLayout(Context context) {
        this(context, (AttributeSet) null);
    }

    public RatioFrameLayout(Context context, AttributeSet attributeSet) {
        this(context, attributeSet, 0);
    }

    public RatioFrameLayout(Context context, AttributeSet attributeSet, int i) {
        this(context, attributeSet, i, 0);
    }

    public RatioFrameLayout(Context context, AttributeSet attributeSet, int i, int i2) {
        super(context, attributeSet, i, i2);
        TypedArray obtainStyledAttributes = context.obtainStyledAttributes(attributeSet, R.styleable.RatioFrameLayout);
        String string = obtainStyledAttributes.getString(R.styleable.RatioFrameLayout_sizeRatio);
        if (!TextUtils.isEmpty(string)) {
            String[] split = string.split(":");
            int length = split.length;
            if (length == 1) {
                this.mWidthRatio = Float.parseFloat(split[0]);
                this.mHeightRatio = 1.0f;
            } else if (length == 2) {
                this.mWidthRatio = Float.parseFloat(split[0]);
                this.mHeightRatio = Float.parseFloat(split[1]);
            } else {
                throw new IllegalArgumentException("are you ok?");
            }
        }
        obtainStyledAttributes.recycle();
    }

    /* access modifiers changed from: protected */
    public void onMeasure(int i, int i2) {
        if (!(this.mWidthRatio == 0.0f || this.mHeightRatio == 0.0f)) {
            float sizeRatio = getSizeRatio();
            ViewGroup.LayoutParams layoutParams = getLayoutParams();
            int mode = View.MeasureSpec.getMode(i);
            int size = View.MeasureSpec.getSize(i);
            int mode2 = View.MeasureSpec.getMode(i2);
            int size2 = View.MeasureSpec.getSize(i2);
            if (layoutParams.width != -2 && layoutParams.height != -2 && mode == 1073741824 && mode2 == 1073741824) {
                float f = (float) size;
                float f2 = f / sizeRatio;
                float f3 = (float) size2;
                if (f2 <= f3) {
                    i2 = View.MeasureSpec.makeMeasureSpec((int) f2, BasicMeasure.EXACTLY);
                } else {
                    float f4 = f3 * sizeRatio;
                    if (f4 <= f) {
                        i = View.MeasureSpec.makeMeasureSpec((int) f4, BasicMeasure.EXACTLY);
                    }
                }
            } else if (layoutParams.width != -2 && mode == 1073741824 && mode2 != 1073741824) {
                i2 = View.MeasureSpec.makeMeasureSpec((int) (((float) size) / sizeRatio), BasicMeasure.EXACTLY);
            } else if (!(layoutParams.height == -2 || mode2 != 1073741824 || mode == 1073741824)) {
                i = View.MeasureSpec.makeMeasureSpec((int) (((float) size2) * sizeRatio), BasicMeasure.EXACTLY);
            }
        }
        super.onMeasure(i, i2);
    }

    public float getWidthRatio() {
        return this.mWidthRatio;
    }

    public float getHeightRatio() {
        return this.mHeightRatio;
    }

    public float getSizeRatio() {
        return this.mWidthRatio / this.mHeightRatio;
    }

    public void setSizeRatio(float f, float f2) {
        this.mWidthRatio = f;
        this.mHeightRatio = f2;
        requestLayout();
        invalidate();
    }
}

File: decompiled/sources/com/hjq/widget/layout/RatioFrameLayout.kt
package com.hjq.widget.layout

import android.content.Context
import android.content.res.TypedArray
import android.text.TextUtils
import android.util.AttributeSet
import android.view.View
import android.view.ViewGroup
import android.widget.FrameLayout
import androidx.constraintlayout.solver.widgets.analyzer.BasicMeasure
import com.hjq.widget.R

class RatioFrameLayout : FrameLayout {
    private var mHeightRatio = 0f
    private var mWidthRatio = 0f

    constructor(context: Context) : this(context, null)

    constructor(context: Context, attrs: AttributeSet?) : this(context, attrs, 0)

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : this(
        context,
        attrs,
        defStyleAttr,
        0
    )

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int, defStyleRes: Int) : super(
        context,
        attrs,
        defStyleAttr,
        defStyleRes
    ) {
        val typedArray: TypedArray = context.obtainStyledAttributes(attrs, R.styleable.RatioFrameLayout)
        val ratio = typedArray.getString(R.styleable.RatioFrameLayout_sizeRatio)
        if (!TextUtils.isEmpty(ratio)) {
            val split = ratio!!.split(":")
            when (split.size) {
                1 -> {
                    mWidthRatio = split[0].toFloat()
                    mHeightRatio = 1f
                }

                2 -> {
                    mWidthRatio = split[0].toFloat()
                    mHeightRatio = split[1].toFloat()
                }

                else -> {
                    throw IllegalArgumentException("are you ok?")
                }
            }
        }
        typedArray.recycle()
    }

    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        var widthSpec = widthMeasureSpec
        var heightSpec = heightMeasureSpec
        if (mWidthRatio != 0f && mHeightRatio != 0f) {
            val sizeRatio = getSizeRatio()
            val params = layoutParams
            val widthMode = View.MeasureSpec.getMode(widthSpec)
            val widthSize = View.MeasureSpec.getSize(widthSpec)
            val heightMode = View.MeasureSpec.getMode(heightSpec)
            val heightSize = View.MeasureSpec.getSize(heightSpec)
            if (
                params.width != ViewGroup.LayoutParams.WRAP_CONTENT &&
                params.height != ViewGroup.LayoutParams.WRAP_CONTENT &&
                widthMode == View.MeasureSpec.EXACTLY &&
                heightMode == View.MeasureSpec.EXACTLY
            ) {
                val width = widthSize.toFloat()
                val targetHeight = width / sizeRatio
                val height = heightSize.toFloat()
                if (targetHeight <= height) {
                    heightSpec = View.MeasureSpec.makeMeasureSpec(targetHeight.toInt(), BasicMeasure.EXACTLY)
                } else {
                    val targetWidth = height * sizeRatio
                    if (targetWidth <= width) {
                        widthSpec = View.MeasureSpec.makeMeasureSpec(targetWidth.toInt(), BasicMeasure.EXACTLY)
                    }
                }
            } else if (
                params.width != ViewGroup.LayoutParams.WRAP_CONTENT &&
                widthMode == View.MeasureSpec.EXACTLY &&
                heightMode != View.MeasureSpec.EXACTLY
            ) {
                heightSpec = View.MeasureSpec.makeMeasureSpec(
                    (widthSize.toFloat() / sizeRatio).toInt(),
                    BasicMeasure.EXACTLY
                )
            } else if (
                params.height != ViewGroup.LayoutParams.WRAP_CONTENT &&
                heightMode == View.MeasureSpec.EXACTLY &&
                widthMode != View.MeasureSpec.EXACTLY
            ) {
                widthSpec = View.MeasureSpec.makeMeasureSpec(
                    (heightSize.toFloat() * sizeRatio).toInt(),
                    BasicMeasure.EXACTLY
                )
            }
        }
        super.onMeasure(widthSpec, heightSpec)
    }

    fun getWidthRatio(): Float {
        return mWidthRatio
    }

    fun getHeightRatio(): Float {
        return mHeightRatio
    }

    fun getSizeRatio(): Float {
        return mWidthRatio / mHeightRatio
    }

    fun setSizeRatio(widthRatio: Float, heightRatio: Float) {
        mWidthRatio = widthRatio
        mHeightRatio = heightRatio
        requestLayout()
        invalidate()
    }
}
